vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -234,32 +243,26 @@
       name: "{{ item.name }}"
       kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: crd
-    vars:
-      query: "[?name == 'v1alpha3'].schema.openAPIV3Schema.description"
-    failed_when: crd.resources[0].spec.versions | json_query(query) is not search("{{ item.search }}")
+    failed_when: upgraded_capi_version not in (crd.resources[0].spec.versions | map(attribute='name'))
     loop:
       - name: machines.cluster.x-k8s.io
-        search: upgradedMachine
       - name: kubeadmcontrolplanes.controlplane.cluster.x-k8s.io
-        search: upgradedKubeadmControlPlane
       - name: kubeadmconfigs.bootstrap.cluster.x-k8s.io
-        search: upgradedKubeadmConfig
 
-  # Due to https://github.com/ansible-collections/kubernetes.core/issues/17
-  # we cannot use k8s_cluster_info. A fix is to be included in 2.0.0 of kubernetes.core
-  # - name: Verify upgraded API resource for Metal3Clusters
-  #   kubernetes.core.k8s_cluster_info:
-  #   register: api_status
-  #   # failed_when: api_status.apis not contains the upgraded m3c resource(s)?
+  - name: Verify upgraded API resources for CAPM3
+    kubernetes.core.k8s_cluster_info:
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: api_status
+    failed_when: "'infrastructure.cluster.x-k8s.io/v1beta1' not in api_status.apis"
 
-  - name: Verify upgraded API resource for Metal3Clusters
+  - name: Verify upgraded CRDs for Metal3Clusters
     kubernetes.core.k8s_info:
       api_version: apiextensions.k8s.io/v1
       kind: CustomResourceDefinition
       name: metal3clusters.infrastructure.cluster.x-k8s.io
       kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: crd
-    failed_when: '"upgm3c" not in crd.resources[0].spec.names.shortNames'
+    failed_when: UPGRADED_CAPM3_VERSION not in (crd.resources[0].spec.versions | map(attribute='name'))
```

Comment:
Now that we can verify the API resources for CAPM3, I think we could actually remove the "Verify upgraded CRDs for Metal3Clusters". This is how we used to do it [before the k8s module](https://github.com/metal3-io/metal3-dev-env/blob/56c32374c4c5713b2d272274e6df44de61aa2b94/vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml#L207). Then we checked the CRDs for CAPI and API resources for CAPM3.

I would even suggest that we check only API resources for both CAPI and CAPM3 now that they actually change from v1alphaX to v1betaX so there is a change that we can detect. The reason for checking the CRDs (from what I understand) was that we made changes to them to *simulate* a change but still kept the "contract" version the same. By checking the API version instead of the CRD we would reduce the spam in the log quite a lot (the CRD is 1000s of lines, the API resources are just ~10).
WDYT? I could do a follow up PR with this later to stick to the point here.